### PR TITLE
no error when the column desc is undefined

### LINKED ISSUE #466: BUG- Rebuild property not working as expected
**Description**
Changing the chart will sometimes not correctly  update the chart without refreshing it even with _rebuild_ set to True. 

**How to reproduce**

### First example
```python
from taipy.gui import Gui
import pandas as pd

all_data = pd.DataFrame({
    "x" : [1, 2, 3, 4],
    "y" : [1, 2, 3, 4],
    "category": ["dog", "cat", "cat", "dog"],
    "bubble_size": [10, 50, 50, 10],
    "color": ["red", "green", "green", "red"]
})

categories = list(all_data["category"].unique())
selected_category = "dog"
data = all_data[all_data["category"] == selected_category]

marker = {
    "color": "color",
    "size": "bubble_size"
}

def on_change_cat(state):
    state.data = all_data[all_data["category"] == state.selected_category].reset_index(drop=True)

md = """
<|{selected_category}|selector|lov={categories}|dropdown|label=Category|on_change=on_change_cat|>
<|{data}|chart|rebuild|mode=markers|x=x|y=y|marker={marker}|>
"""


Gui(md).run()
```

- Change the selector to the other option
- Change it back to the first
- refresh

### Second example

```python
from taipy.gui import Gui
import pandas as pd

data1 = pd.DataFrame({
    "questions" : ["a", "b", "c"],
    "answer1": [2, 4, 5],
    "answer2": [1, 1, 1],
})

data2 = pd.DataFrame({
    "questions" : ["a", "b", "c"],
    "answer3": [7, 9, 8],
    "answer4": [1, 1, 1],
})

data = data1
data_columns = ["answer1", "answer2"]

layout = {
    "barmode": "stack"
}

def change_data(state):
    state.data = data2
    state.data_columns = ["answer3", "answer4"]

md = """
<|Button Label|button|on_action=change_data|>
<|{data}|chart|rebuild|type=bar|x={data_columns}|y=questions|name={data_columns}|layout={layout}|orientation=h|>
"""


Gui(md).run()
```

- Click on the button
- Refresh

**Expected behavior**
Charts should be correctly displayed without having to refresh the page.

**Runtime environment**
Taipy 2.4

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
- add a message to let the user know that he needs to bind data variables
resolves #466

### CODE REVIEW

File: taipy/gui/_renderers/builder.py [hunk 1]
```diff
@@ -775,6 +775,8 @@ def set_value_and_default(
                 else:
                     self.__set_default_value(var_name, var_type=var_type)
         else:
+            if var_type == PropertyType.data:
+                _warn(f"{self.__control_type}.data property should be binded.")
```
Comment (FabienLelaquais) [inline taipy/gui/_renderers/builder.py]:
```suggestion
                _warn(f"{self.__control_type}.data property should be bound.")
```
As usual.

File: frontend/taipy-gui/src/components/Taipy/Chart.tsx [hunk 2]
```diff
@@ -419,7 +421,8 @@ const Chart = (props: ChartProp) => {
                           getArrayValue(config.names, idx) ||
                           (config.columns[trace[1]] ? getColNameFromIndexed(config.columns[trace[1]].dfid) : undefined),
                   } as Record<string, unknown>;
-                  ret.marker = getArrayValue(config.markers, idx, ret.marker || {});
+                  const theMarker = getArrayValue(config.markers, idx, ret.marker || {});
+                  ret.marker = theMarker ? {...theMarker}: theMarker;
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Chart.tsx]:
```suggestion
                  ret.marker = theMarker ? {...theMarker} : {};
```
Easier to read?

**Review by FabienLelaquais — DISMISSED** [review]
👍

File: taipy/gui/_renderers/builder.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/_renderers/builder.py]: Can't put this into my head :-(

**Review by FabienLelaquais — APPROVED** [review]
👍👍

**Review by dinhlongviolin1 — APPROVED** [review]
Tx Fred

**Pull request merged**